src/error.ts

```diff
@@ -15,6 +15,20 @@ import { MIMES } from './utils'
  */
 export class H3Error extends Error {
   static __h3_error__ = true
+  toJSON () {
+    const obj: Pick<H3Error, 'message' | 'statusCode' | 'statusMessage' | 'data'> = {
+      message: this.message,
+      statusCode: this.statusCode
+    }
+
+    if (this.statusMessage) { obj.statusMessage = this.statusMessage }
+    if (this.data && (typeof this.data !== 'object' || Object.keys(this.data).length)) {
```

Comment:
(empty object might have meaning. It is probably an issue with ohmyfetch we should fix)

```suggestion
    if (this.data) {
```